app/src/main/java/com/arctouch/publictransportation/domain/generic/params/RouteIdParam.java

```diff
@@ -0,0 +1,11 @@
+package com.arctouch.publictransportation.domain.generic.params;
+
+import com.arctouch.publictransportation.domain.RouteId;
+
+public class RouteIdParam extends GenericParam<RouteId> {
+
+    public RouteIdParam(int routeId) {
+        super.setParams(new RouteId(new Integer(routeId)));
```

Comment:
the _new Integer_ part is unnecessary - in general, you should leave that kind of boxing (int -> Integer) to the compile instead
```
super.setParams(new RouteId(routeId))